test/test_dynamic_shapes.py

```diff
@@ -426,6 +427,14 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
 
         self.assertTrue(sym_int_encountered)
 
+    @skipIfNoSympy
+    def test_deepcopy(self):
+        shape_env = ShapeEnv()
+        a0 = create_symint(shape_env, 2)
+        assert a0 < 4
+        new_shape_env = copy.deepcopy(shape_env)
+        self.assertEqual(len(new_shape_env.guards), 1)
```

Comment:
worth validating any other properties of ShapeEnv here or is this sufficient?

Reply:
Well, we're using the default pickle implementation for the class, so not sure if there's anything else to test. Let's wait for bug reports I guess.